src/resolver/environment.cc

```diff
@@ -129,6 +129,25 @@ Environment::groupIsLocked( const InstallDescriptors & group,
 
   for ( auto & [iid, descriptor] : group )
     {
+      /* Check for upgrades. */
```

Comment:
non-blocking educational note:
 
For `std::variant` a `std::get_if` like this is totally fine, no need to change it.
A _preferred_ approach is using a `switch { case: ... }` style construct which ensures that you've _really_ checked all variant types.
```c++
class Empty {};
std::variant<int, Empty> value = 1;
std::visit( overloaded {
  []( int x ) { std::cout << "Integer: " << x << std::endl; },
  []( Empty e ) { std::cout << "Empty" << std::endl; }
}, value );
```

The definition of `overloaded` will trigger compilation warnings/errors if you've skipped any variant types.